packages/@react-stately/toast/src/useToastState.ts

```diff
@@ -157,23 +161,28 @@ export class ToastQueue<T> {
       this.queue.splice(index, 1);
     }
 
-    this.updateVisibleToasts();
+    this.updateVisibleToasts(index);
   }
 
   /** Removes a toast from the visible toasts after an exiting animation. */
   remove(key: string) {
+    let index = this.queue.findIndex(t => t.key === key);
     this.visibleToasts = this.visibleToasts.filter(t => t.key !== key);
-    this.updateVisibleToasts();
+    this.updateVisibleToasts(index);
   }
 
-  private updateVisibleToasts() {
+  private updateVisibleToasts(oldIndex = -1) {
     let toasts = this.queue.slice(0, this.maxVisibleToasts);
     if (this.hasExitAnimation) {
       let prevToasts: QueuedToast<T>[] = this.visibleToasts
-        .filter(t => !toasts.some(t2 => t.key === t2.key))
-        .map(t => ({...t, animation: 'exiting'}));
+          .filter(t => !toasts.some(t2 => t.key === t2.key))
+          .map(t => ({...t, animation: 'exiting'}));
+
+      if (oldIndex !== -1) {
+        toasts.splice(oldIndex, 0, prevToasts?.[0]);
+      }
 
-      this.visibleToasts = prevToasts.concat(toasts).sort((a, b) => b.priority - a.priority);
```

Comment:
So technically this updateVisibleToast function would work even if multiple toasts were being added/removed at the same time but now we use an index value so it assumes that we are only adding/removing one at a time right? I was a bit concerned but I guess the add/remove have always only supported adding one at a time anyways. Lemme know if I'm following these changes correctly

Reply:
The stately API currently exposes add, remove, close of one item at a time. When I add a whole bunch with five second timeouts quickly or try to close them all with the close button this change works.

The reason this code was modified is because it was moving closed and removed Toasts to the bottom of the visible queue then doing the exit animation. When you have one item showing and being removed that works, but with multiple visible items and one is removed, the removed item jumps to the bottom of the visible list. We need to keep the item being removed in the current position.